torch/_export/logical_schema.py

```diff
@@ -267,14 +272,11 @@ class Graph:
     nodes: List[Node]
 
     # Tensor values that appear in the graph
-    # They could be graph inputs, graph outputs, or intermediate values produced by nodes
-    # The key is a unique identifider name for the IValue
-    # The name will be used in the graph and node to refer to the IValue
-    ivalues: Dict[str, IValue]
+    # They could be graph inputs, graph outputs, or intermediate tensor values produced by nodes
+    tensor_values: List[TensorValue]
```

Comment:
I intentionally choose to use a list, instead of using a map. 

TensorValue should have a "name" field as the identifier, so that the structure's semantics is complete. As a result, it would be duplicated to store "name" again in map's key.

Reply:
this is fine, but my 2c is that in practice everyone will immediately make a mapping from name => TensorValue, so it seems like you might as well do it here (and get the uniqueness property as a result)

Reply:
Yes, I would imagine runtime will immediately converting this to a map during import, and its easily enough to do that. 

My main argument is that Value's most important property is a "named placeholder": TensorValue might not have metadata populated, but it must have a name. For this reason, "name" should be a field for "Value.

Reply:
oh, I'm saying it's even fine to duplicate it, like: `{"my_tensor": TensorValue(name="my_tensor")}`. Yes it's  denormalized and thus a little less theoretically clean, but imo we should skew on the side of making things easier to implement ("[worse is better](https://www.jwz.org/doc/worse-is-better.html)" and all that).

Either way is fine by me tho